packages/hooks/use-mounted-state/src/useMountedState.tsx

```diff
@@ -0,0 +1,17 @@
+import { useCallback, useEffect, useRef } from 'react'
+
+export function useMountedState(): () => boolean {
+  const mountedRef = useRef<boolean>(false)
+
+  const get = useCallback(() => mountedRef.current, [])
```

Comment:
Question, why return a function rather than a boolean directly?

Reply:
yes, the intention is to use it in any place calling directly where you need, not infering the state calculated in the resolved hook placement

Reply:
this is taken from react-use library

Reply:
👍 , but do we need to re-rewrite it and expose it then? Couldn't we use the one from the react-use lib instead?

Reply:
this are kind of primitives, we should provide and maintain set of common tools  for repetitive behaviors in components.

Reply:
this is the simplest one, its just to add the useCombinedState, which uses it.